### PR TITLE
Ubuntu LTS support

### PR DESCRIPTION
**Pull request by Ecsy**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | yes
| Related tickets | fixes https://github.com/banzaicloud/pke/issues/41
| License         | Apache 2.0

### What's in this PR?

- Ubuntu LTS support
- Cilium Kubernetes CNI network driver
- Bumped Kubernetes default version to v1.16.0
- Fix leaking file descriptors

### Checklist

- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

File: docs/vagrant.md [hunk 1]
```diff
@@ -37,15 +36,14 @@ You can use `kubectl` from now on. Test it by executing `kubectl get nodes`
 
 ### Multi node PKE
 
-Start a machine with the following command: `vagrant up node1 node2`
+Start a machine with the following command: `vagrant up centos1 centos2`
```
Comment (orymate) [inline docs/vagrant.md]: this should be changed in the pipeline quickstart guide as well

File: Vagrantfile [hunk 2]
```diff
@@ -54,4 +54,45 @@ EOF
...
+                echo 'sync time'
+                systemctl start ntp
+                systemctl enable ntp
+
+                echo 'set host name resolution'
+                cat >> /etc/hosts <<EOF
+192.168.64.21 ubuntu1
+192.168.64.22 ubuntu2
+192.168.64.23 ubuntu3
+192.168.64.24 ubuntu4
```
Comment (orymate) [inline Vagrantfile]: the centos nodes should be here too, shouldn't they?

File: docs/vagrant.md (hunk 1 above)
Comment (Ecsy) [reply docs/vagrant.md]: Once this got merged, sure.

File: Vagrantfile (hunk 2 above)
Comment (Ecsy) [reply Vagrantfile]: Not intent to use mixed node setup.

**Comment by denisgolius** [comment]
@Ecsy As I see it supports Ubuntu in clouds now not Vagrant?

**Pull request merged**